content/articles/Async-js-using-ajax/index.md

```diff
@@ -0,0 +1,274 @@
+      <input id="number-of-jokes" type="number" id="number" />
+    </div>
+
+    <div>
+      <button id="generate-jokes">Generate Jokes</button>
+    </div>
+  </form>
+</div>
+```
+
+- Create an ordered list (`ol`) for the jokes to render below the form.
+- Create a JavaScript file and save it as `index.js`.
+- Include the `index.js`.
+
+- Your html file should look like this 👇:
+
+```html
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+
+    <!-- Skeleton CSS -->
+    <link
+      rel="stylesheet"
+      href="https://cdnjs.cloudflare.com/ajax/libs/skeleton/2.0.4/skeleton.css"
+      integrity="sha512-5fsy+3xG8N/1PV5MIJz9ZsWpkltijBI48gBzQ/Z2eVATePGHOkMIn+xTDHIfTZFVb9GMpflF2wOWItqxAP2oLQ=="
+      crossorigin="anonymous"
+      referrerpolicy="no-referrer"
+    />
+
+    <title>Getting Jokes from External API</title>
+  </head>
+  <body>
+    <div class="container">
+      <h2>Asynchronous JS Joke Generator</h2>
+      <form>
+        <div>
+          <label for="number"> Number of jokes to generate</label>
+          <input id="number-of-jokes" type="number" id="number" />
+        </div>
+
+        <div>
+          <button id="generate-jokes">Generate Jokes</button>
+        </div>
+      </form>
+      <ol id="jokes"></ol>
+    </div>
+
+    <script src="./index.js"></script>
+  </body>
+</html>
+```
+
+### The JavaScript
+- Open `index.js`.
+- Grab the button from the DOM.
+
+```javascript
+const button = document.querySelector("#generate-jokes");
+```
+
+- Add an event listener to the button to listen for a click.
+
+```javascript
+button.addEventListener("click", generateNewJokes);
+```
+
+### Prepare the AJAX request
+- Create a function named `generateNewJokes()` for the click event.
+- Grab the `#number-of-jokes` element from the UI and assign it to a variable.
+- Create a variable called request and set it to a new [XHR](https://en.wikipedia.org/wiki/XMLHttpRequest) object.
+
+```javascript
+function generateNewJokes(e) {
+  const newXHRRequest = new XMLHttpRequest();
+  const numberOfJokes = document.querySelector("#number-of-jokes").value;
+}
+```
+
+> "**_XMLHttpRequest (XHR)_** objects are used to interact with servers. You can retrieve data from a URL without having to do a full page refresh. This enables a Web page to update just part of a page without disrupting what the user is doing" - [MDN](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest).
+
+- Inside the `generateNewJokes` function, create the object, `newXHRRequest.open()` that takes three parameters:
+
+  1. The `newXHRRequest`, which is [GET method](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods)
+  2. The link and append the number variable to get the numbers entered by the user and pass it to the.
+  3. Set the last variable to true to make it asynchronous
+
```

Comment:
@gmcodes20 The 2nd point in this list is quite broken. Please fix it.

Reply:
Done. 
I had to rewrite the points, I realized I missed something.